modules/entity/spec/utils.spec.ts

```diff
@@ -0,0 +1,32 @@
+import * as ngCore from '@angular/core';
+import { selectIdValue } from '../src/utils';
+import { BookModel, AClockworkOrange } from './fixtures/book';
+
+describe('Entity utils', () => {
+  describe(`selectIdValue()`, () => {
+    it('should not warn when key does exist', () => {
+      const spy = spyOn(console, 'warn');
+
+      const key = selectIdValue(AClockworkOrange, book => book.id);
+
+      expect(spy).not.toHaveBeenCalled();
+    });
+
+    it('should warn when key does not exist in dev mode', () => {
+      const spy = spyOn(console, 'warn');
+
+      const key = selectIdValue(AClockworkOrange, (book: any) => book.foo);
+
+      expect(spy).toHaveBeenCalled();
+    });
+
+    it('should not warn when key does not exist in prod mode', () => {
+      spyOn(ngCore, 'isDevMode').and.returnValue(false);
```

Comment:
You think it would be better to just set prod mode here and reset it after the `selectIdValue` function call?

Reply:
You mean with `enableProdMode()`? Because this threw an error because the platform was already set.

Reply:
Ok. Just didn't want to use the wildcard import if we didn't have to.